### ISSUE TITLE
Allow defaulting a non-existent path in VRL

### ISSUE THREAD

**Issue by jszwedko** [timberio/vector#7052]
Requested by user in discord: https://discord.com/channels/742820443487993987/746070591097798688/829788575234523156

They'd like to be able to lookup a field but set a default if it does not exist. 

This is doable via:

```
$ bar = if exists(.foo) { .foo } else { "not set" }
"not set"
```

But it would be nice to add some sort of null coalescing support similar to the error coalescing we currently have using `??`.

**Comment by binarylogic** [comment]
Yes, we've discussed null coalescing and I see no reason not to add it. Doesn't a non-existent field error though?

**Comment by jszwedko** [comment]
> Yes, we've discussed null coalescing and I see no reason not to add it. Doesn't a non-existent field error though?

Currently it just returns `null`. That is something we could revisit too though.

**Comment by JeanMertz** [comment]
You _could_ write `bar = .foo || "not set"`, but that would also resolve to the rhs expression if `.foo` is `false`.

If we were to add this, would you have any suggestions on the symbol to use for this use-case @jszwedko?

**Comment by jszwedko** [comment]
Hmm, yeah, that is useful but a bit different.

Maybe we could have a `get()` function that takes a default to return if unnset?
